src/axes.jl

```diff
@@ -96,13 +99,12 @@ IdOffsetRange(r::AbstractUnitRange{T}, offset::Integer = 0) where T<:Integer =
 IdOffsetRange{T,I}(r::IdOffsetRange{T,I}) where {T<:Integer,I<:AbstractUnitRange{T}} = r
 function IdOffsetRange{T,I}(r::IdOffsetRange, offset::Integer = 0) where {T<:Integer,I<:AbstractUnitRange{T}}
     rc, offset_rc = offset_coerce(I, r.parent)
-    return IdOffsetRange{T,I}(rc, r.offset + offset + offset_rc)
+    return IdOffsetRange{T,I}(rc, convert(T, r.offset + offset + offset_rc))
```

Comment:
```suggestion
    return IdOffsetRange{T,I}(rc, convert(T, r.offset + offset + offset_rc)::T)
```
Most of the time that's a no-op, but if anyone supplies a funky `Integer` and then writes a broken `convert` method that doesn't actually return a `T` (there is nothing preventing that from happening), your version would yield a StackOverflowError. This yields a TypeError, which is more graceful.